### PR TITLE
Remove plugin-only option from secrets

### PR DESCRIPTION
**Pull request by anbraten**
- remove plugin-only option from secrets
The images list of secrets is currently pretty use-less if not enabled with the plugin-only option in combination, therefore the plugin-only option will now be automatically set, if at least one image was listed

- rename secrets json variables (images and events) to their proper plural

### CODE REVIEW

File: cli/secret/secret_add.go [hunk 1]
```diff
@@ -46,17 +46,13 @@ var secretCreateCmd = &cli.Command{
 			Usage: "secret value",
 		},
 		&cli.StringSliceFlag{
-			Name:  "event",
```
Comment (6543) [inline cli/secret/secret_add.go:49]: could we use alias to still have old one too?

Comment (anbraten) [reply cli/secret/secret_add.go:49]: I marked this PR as breaking and added a migration note to the docs page.

**Comment by anbraten** [comment]
added a migration

**Comment by lafriks** [comment]
Migration should also remove image value if plugin only is not selected otherwise it will stop functioning if image is set but it's not plugin

**Comment by anbraten** [comment]
> Migration should also remove image value if plugin only is not selected otherwise it will stop functioning if image is set but it's not plugin

Actually while writing the migration note and thinking about this again, I would say it more secure to fail in that case than potentially still leaking the secret. As the user would notice the error of the missing secret I guess its fine to limit existing image filters to plugins.

File: docs/docs/91-migrations.md [hunk 2]
```diff
@@ -8,6 +8,7 @@ Some versions need some changes to the server configuration or the pipeline conf
 - Dropped deprecated `pipeline:` keyword in favor of `steps:` in pipeline config
 - Dropped deprecated `branches:` filter in favor of global [`when.branch`](./20-usage/20-workflow-syntax.md#branch-1) filter
 - Deprecated `platform:` filter in favor of `labels:`, [read more](./20-usage/20-workflow-syntax.md#filter-by-platform)
+- Secrets `event` property was renamed to `events` as its a list of events. The new property `events` has to be used in the api and as cli argument. The old property `event` was removed.
```
[the remark this answers is not in the document]
Comment (anbraten) [reply docs/docs/91-migrations.md:11]: added

**Pull request merged**